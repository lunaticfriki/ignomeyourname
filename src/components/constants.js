// header
const headerTitle = 'i gnome your name!'
const headerSubtitle = 'Search your favorite gnome from Brastlewark!'

//nav
const navSearcher = 'Searcher'
const navAbout = 'About me'

//main
const mainTitle = ''
const mainubtitle = ''

// searcher
const sideSearcherTitle = 'SEARCHER'

//footer
const footerSignature = 'oscar ventura -'
const footerMessage = 'gnome searcher'

export {
  headerTitle,
  headerSubtitle,
  navSearcher,
  navAbout,
  mainTitle,
  mainubtitle,
  sideSearcherTitle,
  footerSignature,
  footerMessage
}
