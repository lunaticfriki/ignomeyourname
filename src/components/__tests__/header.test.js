import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

import Header from '../header'
import { translations } from '../constants'

describe('Header component suite', () => {
  const { headerTitle, headerSubtitle, navSearcher, navAbout, navGallery } = translations
  const wrapperMount = mount(<Header />)
  test('Header snapshot', () => {
    const component = renderer.create(<Header />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('Should render a Header component', () => {
    expect(wrapperMount).toBeTruthy()
  })
  test('Should render the texts', () => {
    expect(wrapperMount.find('h1').text()).toEqual(headerTitle.toUpperCase())
    expect(wrapperMount.find('h2').text()).toEqual(headerSubtitle)
    expect(
      wrapperMount
        .find('li')
        .at(0)
        .text()
    ).toEqual(navSearcher)
    expect(
      wrapperMount
        .find('li')
        .at(1)
        .text()
    ).toEqual(navGallery)
    expect(
      wrapperMount
        .find('li')
        .at(2)
        .text()
    ).toEqual(navAbout)
  })
})
