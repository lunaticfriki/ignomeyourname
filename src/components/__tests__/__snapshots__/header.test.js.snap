// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component suite Header snapshot 1`] = `
<div
  className="sc-AxjAm hgpxJj"
>
  <a
    aria-current="page"
    href="/"
    onClick={[Function]}
    style={
      Object {
        "color": "#d62b9a",
      }
    }
  >
    <h1>
      I GNOME YOUR NAME!
    </h1>
  </a>
  <h2>
    Search your favorite gnome from Brastlewark!
  </h2>
  <nav>
    <a
      aria-current="page"
      href="/"
      onClick={[Function]}
      style={
        Object {
          "color": "#d62b9a",
        }
      }
    >
      <li>
        Searcher
      </li>
    </a>
    <a
      href="/about"
      onClick={[Function]}
      style={
        Object {
          "color": "#fff",
        }
      }
    >
      <li>
        About me
      </li>
    </a>
  </nav>
</div>
`;
